plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayer.java

```diff
@@ -190,15 +190,21 @@ private void setLayerVisibility(String layerId, boolean visible) {
     }
   }
 
-  private void addLayers() {
-    addSymbolLayer(SHADOW_LAYER, BACKGROUND_LAYER);
-    addSymbolLayer(BACKGROUND_LAYER, FOREGROUND_LAYER);
-    addSymbolLayer(FOREGROUND_LAYER, null);
-    addSymbolLayer(BEARING_LAYER, null);
+  private void addLayers(LocationLayerOptions options) {
+    addSymbolLayer(SHADOW_LAYER, BACKGROUND_LAYER, options);
+    addSymbolLayer(BACKGROUND_LAYER, FOREGROUND_LAYER, options);
+    addSymbolLayer(FOREGROUND_LAYER, null, options);
+    addSymbolLayer(BEARING_LAYER, null, options);
     addAccuracyLayer();
   }
 
-  private void addSymbolLayer(String layerId, String beforeLayerId) {
+  private void addSymbolLayer(String layerId, String beforeLayerId, LocationLayerOptions options) {
+    String foregroundIconString = buildIconString(options.foregroundName(), FOREGROUND_ICON);
+    String foregroundStaleIconString = buildIconString(options.foregroundStaleName(), FOREGROUND_STALE_ICON);
+    String backgroundIconString = buildIconString(options.backgroundName(), BACKGROUND_ICON);
+    String backgroundStaleIconString = buildIconString(options.backgroundStaleName(), BACKGROUND_STALE_ICON);
+    String bearingIconString = buildIconString(options.bearingName(), BEARING_ICON);
```

Comment:
I think this values would have to be moved to the location feature, otherwise, you'd not be able to revert this custom image names when resetting the style on an already initialized plugin.

We could also document this behavior and leave as is, thoughts?